Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -305,126 +305,244 @@ final class DocumentTests: MongoSwiftTestCase {
+                    // -> bson_t. At the end, the new bson_t should be identical to the original one. If not, our bson_t
+                    // translation layer is lossy and/or buggy.
+                    let nativeFromDoc = docFromCB.toArray()
+                    let docFromNative = Document(fromArray: nativeFromDoc)
+                    expect(docFromNative.rawBSON).to(equal(cBData))
+
+                    // native_to_canonical_extended_json( bson_to_native(cB) ) = cEJ
+                    expect(docFromCB.canonicalExtendedJSON).to(cleanEqual(test.canonicalExtJSON))
+
+                    // native_to_relaxed_extended_json( bson_to_native(cB) ) = rEJ (if rEJ exists)
+                    if let rEJ = test.relaxedExtJSON {
+                        expect(try Document(fromBSON: cBData).extendedJSON).to(cleanEqual(rEJ))
+                    }
 
-                let cEJ = validCase["canonical_extjson"] as? String ?? ""
-                guard let cEJData = cEJ.data(using: .utf8) else {
-                    XCTFail("Unable to interpret canonical_extjson as Data")
-                    return
-                }
+                    // for cEJ input:
+                    // native_to_canonical_extended_json( json_to_native(cEJ) ) = cEJ
+                    expect(try Document(fromJSON: cEJData).canonicalExtendedJSON).to(cleanEqual(test.canonicalExtJSON))
 
-                let lossy = validCase["lossy"] as? Bool ?? false
-
-                // for cB input:
-                // native_to_bson( bson_to_native(cB) ) = cB
-                let docFromCB = try Document(fromBSON: cBData)
-                expect(docFromCB.rawBSON).to(equal(cBData))
-
-                // test round tripping through documents
-                // We create an array by reading every element out of the document (and therefore out of the bson_t)
-                // We then create a new document and append each element of the array to it. Once that is done, every
-                // element in the original document will have gone from bson_t -> Swift data type -> bson_t. At the end,
-                // the new bson_t should be identical to the original one. If not, our bson_t translation layer is
-                // lossy and/or buggy.
-                let nativeFromDoc = docFromCB.toArray()
-                let docFromNative = Document(fromArray: nativeFromDoc)
-                expect(docFromNative.rawBSON).to(equal(cBData))
-
-                // native_to_canonical_extended_json( bson_to_native(cB) ) = cEJ
-                expect(docFromCB.canonicalExtendedJSON).to(cleanEqual(cEJ))
-
-                // native_to_relaxed_extended_json( bson_to_native(cB) ) = rEJ (if rEJ exists)
-                if let rEJ = validCase["relaxed_extjson"] as? String {
-                    expect(try Document(fromBSON: cBData).extendedJSON).to(cleanEqual(rEJ))
-                }
+                    // native_to_canonical_extended_json( json_to_native(cEJ) ) = cEJ
```

Comment:
this comment doesn't seem to line up with what this is checking. I would expect this to be comparing `Document(fromJSON: cEJData).canonicalExtJSON` with `test.canonicalExtJSON`